codemp/client/cl_main.cpp

```diff
@@ -3045,10 +3078,37 @@ void CL_ServerInfoPacket( netadr_t from, msg_t *msg ) {
 	// if this isn't the correct protocol version, ignore it
 	prot = atoi( Info_ValueForKey( infoString, "protocol" ) );
 	if ( prot != PROTOCOL_VERSION ) {
-		Com_DPrintf( "Different protocol info packet: %s\n", infoString );
+		Com_DPrintf("Different protocol info packet: %s\n", infoString);
 		return;
 	}
 
+	if (!clc.httpdl[0]) {
+		char *val;
+
+		val = Info_ValueForKey(infoString, "mvhttp");
+		if (strtol(val, NULL, 10)) {
+			Com_sprintf(clc.httpdl, sizeof(clc.httpdl), "http://%i.%i.%i.%i:%s",
+						clc.serverAddress.ip[0], clc.serverAddress.ip[1],
+						clc.serverAddress.ip[2], clc.serverAddress.ip[3], val);
+		} else if ((val = Info_ValueForKey(infoString, "mvhttpurl")) && Q_stristr(val, "http://")) {
+			Q_strncpyz(clc.httpdl, val, sizeof(clc.httpdl));
+
+			// make sure there is no "/" on the end
+			// so it always is in the format "http://a.org"
+			size_t len = strlen(clc.httpdl);
+			if (clc.httpdl[len - 1] == '/') {
+				clc.httpdl[len - 1] = 0;
+			}
+		}
+	}
+
+	// if this is an MB2 server, ignore it
+	if (!Q_stricmp(Info_ValueForKey(infoString, "game"), "mbii") && Q_stricmp(Cvar_VariableString("fs_game"), "mbii")) {
+		return;
+	}
```

Comment:
Do we really need to treat MB2 as a special case for base clients?

Reply:
Removed this redundant check